components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/auth/InMemorySessionManager.java

```diff
@@ -0,0 +1,175 @@
+        return session;
+    }
+
+    @Override
+    public Optional<Session> getSession(HttpRequest request, HttpResponse response) {
+        String sessionId = request.getCookieValue(Session.SESSION_COOKIE_NAME);
+        if (sessionId == null) {
+            return Optional.empty();
+        }
+        if (!Session.isValidSessionId(sessionId)) {
+            throw new IllegalArgumentException("Session ID '" + sessionId + "' is invalid.");
+        }
+        String contextPath = request.getContextPath();
+        Cache<String, Session> cache = getCache(contextPath, cacheConfiguration);
+        return Optional.ofNullable(cache.get(sessionId));
+    }
+
+    @Override
+    public boolean destroySession(HttpRequest request, HttpResponse response) {
+        String sessionId = request.getCookieValue(Session.SESSION_COOKIE_NAME);
+        if (sessionId == null) {
+            return true; // Session not available
+        }
+        if (!Session.isValidSessionId(sessionId)) {
+            throw new IllegalArgumentException("Session ID '" + sessionId + "' is invalid.");
+        }
+        String contextPath = request.getContextPath();
+        Cache<String, Session> cache = getCache(contextPath, cacheConfiguration);
+        return cache.remove(sessionId);
+    }
+
+    @Override
+    public void clear() {
+        cacheManager.getCacheNames().forEach(cacheName -> {
+            Cache<String, Session> cache = cacheManager.getCache(cacheName);
+            if (!cache.isClosed()) {
+                cache.clear();
+            }
+        });
+    }
+
+    @Override
+    public int getCount() {
+        int size = 0;
+        for (String cacheName : cacheManager.getCacheNames()) {
+            size += Iterables.size(cacheManager.getCache(cacheName));
+        }
+        return size;
+    }
+
+    @Override
+    public void close() {
+        if (!cacheManager.isClosed()) {
+            cacheManager.getCacheNames().forEach(cacheName -> {
+                Cache<String, Session> cache = cacheManager.getCache(cacheName);
+                if (!cache.isClosed()) {
+                    cache.clear();
+                    cache.close();
+                }
+            });
+        }
+        cacheManager.close();
+    }
+
+    @Override
+    protected void finalize() throws Throwable {
+        close();
+        super.finalize();
+    }
+
+    private Cache<String, Session> getCache(String cacheName, MutableConfiguration<String, Session> cacheConfig) {
+        try {
+            synchronized (LOCK) {
+                Cache<String, Session> cache = cacheManager.getCache(cacheName, String.class, Session.class);
+                if (cache == null) {
+                    cache = cacheManager.createCache(cacheName, cacheConfig);
+                }
+                return cache;
+            }
+        } catch (IllegalStateException e) {
+            String message = new StringBuilder().append("Cannot create cache '").append(cacheName)
```

Comment:
I don't see a necessity for using `StringBuilder` here. `+` operator is more suitable here. Compiler will replace the `+` with `StringBuilder` (see [JLS](https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.18.1))